### PR TITLE
OETH Base deploy script

### PR DESCRIPTION
**Pull request by PraneshASP**
If you made a contract change, make sure to complete the checklist below before merging it in master.

Refer to our [documentation](https://github.com/OriginProtocol/security) for more details about contract security best practices.

Contract change checklist:
  - [ ] Code reviewed by 2 reviewers. 
  - [ ] Copy & paste code review [security checklist](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review-Example.md) below this checklist.
  - [ ] Unit tests pass
  - [ ] Slither tests pass with no warning
  - [ ] Echidna tests pass if PR includes changes to OUSD contract (not automated, run manually on local)

### CODE REVIEW

File: contracts/deploy/base/092_woeth_on_base.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
-const { deployOnBase } = require("../utils/delpoy-l2");
-const { deployWithConfirmation, withConfirmation } = require("../utils/deploy");
-const { getTxOpts } = require("../utils/tx");
+const { deployOnBase } = require("../../utils/delpoy-l2");
+const {
+  deployWithConfirmation,
+  withConfirmation,
+} = require("../../utils/deploy");
+const { getTxOpts } = require("../../utils/tx");
```
Comment (sparrowDom) [inline contracts/deploy/base/092_woeth_on_base.js:6]: You can rename this file to `002_woeth_on_base.js`

**Review by sparrowDom — COMMENTED** [review]
Aside from a minor file rename, the CI seems to not work. Would be cool that before deployment the all the unit & fork tests in the CI succeed

File: contracts/fork-test.sh [hunk 2]
```diff
@@ -80,6 +80,10 @@ main()
             # Run all files with `.holesky.fork-test.js` suffix when no file name param is given
             # pass all other params along
             params+="test/**/*.holesky.fork-test.js"
+        elif [[ $FORK_NETWORK_NAME == "holesky" ]]; then
```
Comment (shahthepro) [inline contracts/fork-test.sh:83]: Should this be `base`?

File: contracts/fork-test.sh [hunk 3]
```diff
@@ -101,7 +105,7 @@ main()
         FORK=true IS_TEST=true npx --no-install hardhat coverage --testfiles "${params[@]}"
     else
         echo "Running fork tests..."
-        FORK=true IS_TEST=true npx --no-install hardhat test ${params[@]}
+        FORK=true IS_TEST=true npx --no-install hardhat test ${params[@]} --show-stack-traces
```
Comment (shahthepro) [inline contracts/fork-test.sh:108]: why this change?

File: contracts/hardhat.config.js [hunk 4]
```diff
@@ -195,6 +205,8 @@ module.exports = {
         process.env.FORK === "true"
           ? isHoleskyFork
             ? HOLESKY_DEPLOYER
+            : isBaseFork
+            ? BASE_DEPLOYER
```
Comment (shahthepro) [inline contracts/hardhat.config.js:209]: I don't agree with this change personally. If we change the governor on fork, we are deploying/testing the contracts in a state that's gonna be completely different from mainnet.

File: contracts/test/_global-hooks.js [hunk 5]
```diff
@@ -34,12 +34,25 @@ mocha.before(function () {
...
-      s.file.endsWith(".arb.fork-test.js") == isArbFork &&
-      s.file.endsWith(".base.fork-test.js") == isBaseFork
-  );
+  root.suites = root.suites.filter((s) => {
+    const isMainnetForkTestFile = s.file.endsWith(".fork-test.js");
+    const isHoleskyTestFile = s.file.endsWith(".holesky-fork-test.js");
+    const isArbTestFile = s.file.endsWith(".arb.fork-test.js");
+    const isBaseTestFile = s.file.endsWith(".base.fork-test.js");
+
+    if (isMainnetForkTest) {
```
Comment (shahthepro) [inline contracts/test/_global-hooks.js:43]: I think I changed the order and fixed it in some PR. `isMainnetForkTestFile` should always be at the end and should always be fully spread like `isMainnetForkTestFile && !isHoleskeyTestFile && !isArbTestFile`

File: contracts/utils/hardhat-helpers.js [hunk 6]
```diff
@@ -5,14 +5,19 @@ const isFork = process.env.FORK === "true";
 const isArbitrumFork = process.env.FORK_NETWORK_NAME === "arbitrumOne";
 const isHoleskyFork = process.env.FORK_NETWORK_NAME === "holesky";
 const isHolesky = process.env.NETWORK_NAME === "holesky";
+const isBase = process.env.NETWORK_NAME === "base";
```
Comment (shahthepro) [inline contracts/utils/hardhat-helpers.js:8]: checking chainId would be much better here

**Review by shahthepro — COMMENTED** [review]
Few comments inline. Random question: Storage layout file isn't updated?

File: contracts/test/_global-hooks.js (hunk 5 above)
Comment (PraneshASP) [reply contracts/test/_global-hooks.js]: Thanks for the fix

File: contracts/fork-test.sh (hunk 3 above)
Comment (PraneshASP) [reply contracts/fork-test.sh]: removed

File: contracts/fork-test.sh (hunk 2 above)
Comment (PraneshASP) [reply contracts/fork-test.sh]: good catch, changed.

File: contracts/hardhat.config.js (hunk 4 above)
Comment (PraneshASP) [reply contracts/hardhat.config.js]: Agreed, this has been removed.

File: contracts/deploy/base/092_woeth_on_base.js (hunk 1 above)
Comment (PraneshASP) [reply contracts/deploy/base/092_woeth_on_base.js]: Done

File: contracts/utils/hardhat-helpers.js (hunk 6 above)
Comment (sparrowDom) [reply contracts/utils/hardhat-helpers.js:8]: Yup agreed, much safer
